patches/ui-webui-resources-tools-build_webui.gni.patch

```diff
@@ -1,28 +1,84 @@
-@@ -190,6 +190,7 @@ template("build_webui") {
+@@ -104,12 +104,29 @@ template("build_webui") {
+     }
+   }
+ 
++  # Generate the list of additional outputs generated from overrides
++  override_output_files = []
++  if (defined(invoker.override_files)) {
++    foreach(override, invoker.override_files) {
++      override_output_files += [get_path_info(override, "dir") + "/" +
++                     get_path_info(override, "name") + "-chromium." +
++                     get_path_info(override, "extension") ]
++    }
++  }
++
++  override_ts_files = filter_include(override_output_files, ["*.ts"])
++
+   if (defined(invoker.css_files)) {
+     # Files that are generated by css_to_wrapper().
+     css_wrapper_files = []
+     foreach(f, invoker.css_files) {
+       css_wrapper_files += [ f + ".ts" ]
+     }
++    
++    override_css_files = filter_include(override_output_files, ["*.css"])
++    foreach(f, override_css_files) {
++      css_wrapper_files += [f + ".ts" ]
++    }
+   }
+ 
+   # Generated Mojo JS files.
+@@ -190,6 +207,11 @@ template("build_webui") {
          enable_removal_comments = enable_source_maps
        }
        in_files = ts_files
-+      if (defined(invoker.exclude_ts_preprocess_files)) { in_files -= invoker.exclude_ts_preprocess_files } if (defined(invoker.preprocess_deps)) { public_deps = invoker.preprocess_deps }
++      if (defined(invoker.override_files)) {
++        override_files = filter_include(invoker.override_files + invoker.replace_files, ["*.ts"])
++        override_file_outputs = override_ts_files
++      }
++      if (defined(invoker.exclude_ts_preprocess_files)) { in_files -= invoker.exclude_ts_preprocess_files } if (defined(invoker.preprocess_deps)) { public_deps = invoker.preprocess_deps } 
      }
    }
  
-@@ -240,6 +241,7 @@ template("build_webui") {
+@@ -240,6 +262,11 @@ template("build_webui") {
        if (defined(invoker.css_files)) {
          in_files += invoker.css_files
        }
++      if (defined(invoker.override_files)) {
++        override_files = filter_include(invoker.override_files + invoker.replace_files, ["*.css"])
++        override_file_outputs = override_css_files
++      }
 +      if (defined(invoker.exclude_html_css_preprocess_files)) { in_files -= invoker.exclude_html_css_preprocess_files }
      }
    }
  
-@@ -266,6 +268,7 @@ template("build_webui") {
+@@ -266,6 +293,7 @@ template("build_webui") {
            }
          }
        }
 +      if (defined(invoker.preprocess_deps)) { public_deps = invoker.preprocess_deps }
      }
    }
  
+@@ -275,7 +303,7 @@ template("build_webui") {
+       deps = [ ":preprocess_html_css_files" ]
+       in_folder = preprocess_dir
+       out_folder = preprocess_dir
+-      in_files = invoker.css_files
++      in_files = invoker.css_files + override_css_files
+       minify = optimize
+     }
+   }
+@@ -349,7 +377,7 @@ template("build_webui") {
+       }
+     }
+ 
+-    in_files = ts_files
++    in_files = ts_files + override_ts_files
```

Comment:
I think this can just be `in_files += ...` so it doesn't affect the previous line (and possibly combine this with the previous patch, though I can't quite tell if that's possible looking at just the diff).

Reply:
Unfortunately can't be combined as this is a separate target